test/adapters/unit/LayerZeroAdapter.t.sol

```diff
@@ -0,0 +1,217 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.28;
+
+import {IAuth} from "../../../src/misc/interfaces/IAuth.sol";
+import {CastLib} from "../../../src/misc/libraries/CastLib.sol";
+
+import {Mock} from "../../common/mocks/Mock.sol";
+
+import {IAdapter} from "../../../src/common/interfaces/IAdapter.sol";
+import {IMessageHandler} from "../../../src/common/interfaces/IMessageHandler.sol";
+
+import "forge-std/Test.sol";
+
+import {LayerZeroAdapter} from "../../../src/adapters/LayerZeroAdapter.sol";
+import {
+    ILayerZeroAdapter,
+    IAdapter,
+    ILayerZeroReceiver,
+    ILayerZeroEndpointV2,
+    MessagingParams,
+    MessagingFee,
+    MessagingReceipt,
+    Origin,
+    LayerZeroSource,
+    LayerZeroDestination
+} from "../../../src/adapters/interfaces/ILayerZeroAdapter.sol";
+
+contract MockLayerZeroEndpoint is Mock {
+    function send(MessagingParams calldata params, address refundAddress)
+        external
+        payable
+        returns (MessagingReceipt memory)
+    {
+        values_uint32["params.dstEid"] = params.dstEid;
+        values_bytes32["params.receiver"] = params.receiver;
+        values_bytes["params.message"] = params.message;
+        values_bytes["params.options"] = params.options;
+        values_bool["params.payInLzToken"] = params.payInLzToken;
+
+        values_address["refundAddress"] = refundAddress;
+
+        return MessagingReceipt(bytes32(""), 0, MessagingFee(0, 0));
+    }
+
+    function setDelegate(address newDelegate) external {
+        values_address["delegate"] = newDelegate;
+    }
+
+    function quote(MessagingParams calldata, address) external pure returns (MessagingFee memory) {
+        // TODO: decode gas limit from params.options
```

Comment:
A reminder of this TODO

Reply:
Yeah I will remove this TODO. It is a bit complex, and not really much value. More valuable to add the integration tests that our auditors wrote to the codebase (https://gist.github.com/gvladika/ad74ef148840879b0b60fba96278a50e)